### PR TITLE
Add metric documentation

### LINKED ISSUE #1468: Add some key metrics on the ingestion
Now that we have an indexing server, we can expose some nice prometheus metrics.

### PR DESCRIPTION
**Pull request by boraarslan**

### CODE REVIEW

File: docs/reference/metrics.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+sidebar_position: 6
+---
+
+Quickwit exposes some key metrics through [Prometheus](https://prometheus.io/). You can use any front-end that supports Prometheus to visually examine the behaviour of Quickwit.
+
+## CLI Metrics
+
+| Namespace | Metric Name | Description | Type |
+| --------- | ----------- | ----------- | ---- |
+| `quickwit` | `allocated_num_bytes` | Number of bytes allocated memory, as reported by jemallocated. | `gauge` |
```
Comment (guilload) [inline docs/reference/metrics.md:12]:
```suggestion
| `quickwit` | `allocated_num_bytes` |  Number of bytes allocated memory, as reported by jemalloc. | `gauge` |
```

File: docs/reference/metrics.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+| `quickwit_storage` | `object_storage_download_num_bytes` | Amount of data downloaded from an object storage. | `counter` |
+
+## Cache Metrics
+
+Currently Quickwit exposes metrics for three caches: `fastfields`, `shortlived`, `splitfooter`. These metrics share the same structure.
+
+| Namespace | Metric Name | Description | Type |
+| --------- | ----------- | ----------- | ---- |
+| `cache_{cache_name}` | `in_cache_count` | Count of {cache_name} in cache | `gauge` |
+| `cache_{cache_name}` | `in_cache_num_bytes` | Number of {component_name} bytes in cache | `gauge` |
```
Comment (guilload) [inline docs/reference/metrics.md:44]: replace with `component_name` with `cache_name`? or the other way around?

**Review by guilload — APPROVED** [review]
Looking good!

**Comment by boraarslan** [comment]
Splitting this PR to two separate ones as suggested by @fmassot

**Pull request merged**